### ISSUE TITLE
[Perf] Higgs TTS: batched vocoder decode

### ISSUE THREAD

**Issue by Ccyest** [sgl-project/sglang-omni#569]
## Motivation

The vocoder stage (`create_vocoder_executor`) currently processes requests one at a time. Even though `SimpleScheduler` collects up to `max_batch_size=4` messages, no `batch_compute_fn` is provided, so it falls back to a serial loop — calling `codec.decode(codes_TN)` once per request.

Each `codec.decode` call converts a single `[T, N]` tensor into `[1, N, T]` and runs the full decoder (dequantize → fc2 → acoustic_decoder ConvTranspose stack). At low batch sizes the GPU is severely underutilized because the ConvTranspose kernels are memory-bound and their launch overhead dominates.

The underlying `HiggsAudioV2TokenizerModel.decode()` already accepts batched input `(batch_size, num_quantizers, codes_length)` — we just don't use it. Padding variable-length code sequences and running a single batched decode should reduce per-request vocoder latency roughly proportional to the batch size.

Qwen3-TTS already implements this pattern via `_vocode_batch` + `batch_compute_fn` in `sglang_omni/models/qwen3_tts/stages.py`.

## Current path (single decode per request)

```python
# sglang_omni/models/higgs_tts/audio_codec.py
def decode(self, codes_TN: torch.Tensor) -> torch.Tensor:
    codes_BNT = codes_TN.transpose(0, 1).unsqueeze(0)          # [1, N, T]
    return self.model.decode(codes_BNT).audio_values.squeeze()  # [L]

# sglang_omni/models/higgs_tts/stages.py — _vocode()
codes_TN = reverse_delay_pattern(delayed_LN)
waveform = codec.decode(codes_TN)                              # one at a time
```

`SimpleScheduler` is instantiated with `max_batch_size=4` but without `batch_compute_fn`, so the scheduler's batch collection is wasted.

## Proposed change

1. Add a `decode_batch` method to `HiggsAudioCodec`:

```python
def decode_batch(self, codes_list: list[torch.Tensor]) -> list[torch.Tensor]:
    """Decode a batch of [T_i, N] code tensors → list of [L_i] waveforms."""
    lengths = [c.shape[0] for c in codes_list]
    T_max = max(lengths)
    # Pad to uniform length, stack → [B, N, T_max]
    padded = torch.stack([
        F.pad(c.transpose(0, 1), (0, T_max - c.shape[0]))
        for c in codes_list
    ]).to(device=self.device, dtype=torch.long)
    # Single batched decode
    wavs_BL = self.model.decode(padded).audio_values.squeeze(1)
    # Trim padding from each waveform
    hop = wavs_BL.shape[-1] // T_max  # samples per codec frame
    return [wavs_BL[i, :lengths[i] * hop].cpu() for i in range(len(codes_list))]
```

2. Add `_vocode_batch` in `stages.py` and wire it into `SimpleScheduler`:

```python
def _vocode_batch(payloads: list[StagePayload]) -> list[StagePayload]:
    items = [_prepare_vocoder_item(p) for p in payloads]
    codes_list = [item.codes_TN for item in items]
    waveforms = codec.decode_batch(codes_list)
    return [_finalize(p, item, wav) for p, item, wav in zip(payloads, items, waveforms)]

return SimpleScheduler(
    _vocode,
    batch_compute_fn=_vocode_batch,
    max_batch_size=max_batch_size,
    max_batch_wait_ms=max_batch_wait_ms,
)
```

This mirrors the Qwen3-TTS `_vocode_batch` pattern in `sglang_omni/models/qwen3_tts/stages.py`.

## Scope

- `sglang_omni/models/higgs_tts/audio_codec.py` — add `decode_batch`
- `sglang_omni/models/higgs_tts/stages.py` — add `_vocode_batch`, wire `batch_compute_fn`
- `tests/unit_test/higgs_tts/` — add test for batched decode numerical parity

## Acceptance Criteria

**Correctness**
- [ ] `decode_batch([codes_TN])` produces bit-identical output to `decode(codes_TN)` for single-item batches
- [ ] Multi-item batches with different lengths produce the same waveforms as individual `decode` calls (padding doesn't leak into output)
- [ ] Existing unit tests under `tests/unit_test/higgs_tts/` still pass

**Performance**
- [ ] Vocoder stage throughput (requests/s) strictly higher than single-decode baseline at concurrency ≥ 2
- [ ] No regression at concurrency = 1 (batch of 1 path should have negligible overhead vs current path)

## Known Pitfalls

**Padding-induced artifacts.** The acoustic decoder is a ConvTranspose stack — padding frames must not produce non-zero samples that bleed into the real waveform region. Trim by codec frame count × hop size, not by a fixed ratio. Verify with an A/B waveform diff that the trimmed output matches the unbatched decode.

**Variable-length memory spike.** If one request in the batch is 10× longer than the rest, padding wastes GPU memory proportionally. Consider capping the max length ratio within a batch (e.g. skip batching if `T_max / T_min > 4`) and falling back to serial decode for outliers. The `request_cost_fn` + `max_batch_cost` parameters on `SimpleScheduler` can enforce this at the scheduler level.

**`max_batch_size` tuning.** The codec decoder is relatively lightweight (~100M params). At `max_batch_size=4` the GPU may still be underutilized. Profile whether increasing to 8 or 16 improves throughput without OOM on the target GPU.

## References

- Qwen3-TTS batched vocoder reference: `sglang_omni/models/qwen3_tts/stages.py` (`_vocode_batch`)
- `SimpleScheduler` batch API: `sglang_omni/scheduling/simple_scheduler.py` (`batch_compute_fn` parameter)
- Higgs codec: `sglang_omni/models/higgs_tts/audio_codec.py` (`HiggsAudioCodec.decode`)
- Underlying model: `HiggsAudioV2TokenizerModel.decode()` — already supports `(batch_size, num_quantizers, T)`
- Tracking issue: #478

**Comment by SandyLuXY** [comment]
Thanks @Ccyest! I'm interested in taking this issue.
